clients/vault/src/system.rs

```diff
@@ -329,11 +329,22 @@ impl VaultService {
 		shutdown: ShutdownSender,
 	) -> Result<Self, Error> {
 		let is_public_network = spacewalk_parachain.is_public_network().await?;
+		let cfg_string = fs::read_to_string(&config.stellar_overlay_config_filepath)?;
+		let cfg: StellarOverlayConfig = serde_json::from_str(&cfg_string)
+			.map_err(|e| Error::ConfigError(format!("{:?}", e)))?;
+
+		let stellar_vault_secret_key = std::str::from_utf8(cfg.secret_key())?;
```

Comment:
Maybe we should actually change the `cfg.secret_key()` function to return a string already?